R/kernel.r

```diff
@@ -101,11 +101,47 @@ handle_shell = function() {
         history(msg)
     } else if (msg$header$msg_type == "complete_request") {
         complete(msg)
+    } else if (msg$header$msg_type == "is_complete_request") {
+        is_complete(msg)
     } else {
         print(c("Got unhandled msg_type:", msg$header$msg_type))
     }
 },
 
+is_complete = function(request) {
+    code <- request$content$code
+    message <- tryCatch({
+        evaluate:::parse_all(code)
```

Comment:
this is [exported in `evaluate`](https://github.com/hadley/evaluate/blob/master/R/parse.r#L11), so why three “:::”? does this even work? i thought exported stuff requires exactly two “::”?